### PR TITLE
Extend ipi-metal test coverage

### PR DESCRIPTION
**Pull request by akiselev1**

### CODE REVIEW

**Comment by andfasano** [comment]
How long does it takes to execute this set? To avoid hitting all the other repos, could make sense to implement a solution similar to the one adopted for injecting the configuration, so that we could build a single dedicated job verifying VZ cases until deciding to promote them as default?

**Comment by akiselev1** [comment]
@andfasano It takes 28 minutes. Maybe we can leave the existing e2e-metal-ipi as is to keep it relatively fast and create a new e2e-metal-ipi-extended? Either way, I will look into injecting test cases approach.

File: ci-operator/step-registry/baremetalds/e2e/baremetalds-e2e-workflow.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ workflow:
     - ref: baremetalds-packet-setup
     - ref: ipi-install-rbac
     - ref: baremetalds-devscripts-setup
+    - ref: baremetalds-e2e-conf
```
Comment (andfasano) [inline ci-operator/step-registry/baremetalds/e2e/baremetalds-e2e-workflow.yaml:8]: I think it's safe to move this step into the `test` section

File: ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh [hunk 2]
```diff
@@ -34,8 +34,18 @@ scp "${SSHOPTS[@]}" /usr/bin/openshift-tests /usr/bin/kubectl "root@${IP}:/usr/l
...
 set +e
-echo "### Running tests"
-ssh "${SSHOPTS[@]}" "root@${IP}" openshift-tests run "openshift/conformance/parallel" --dry-run \| grep 'Feature:ProjectAPI' \| openshift-tests run -o /tmp/artifacts/e2e.log --junit-dir /tmp/artifacts/junit -f -
+if [[ -s "${SHARED_DIR}/test-list" ]]; then
+    echo "### Copying test-list file"
+    scp "${SSHOPTS[@]}" "${SHARED_DIR}/test-list" "root@${IP}:/tmp/test-list"
+    echo "### Checking test-list file"
+    ssh "${SSHOPTS[@]}" "root@${IP}" ls -l /tmp/test-list 
+    ssh "${SSHOPTS[@]}" "root@${IP}" cat /tmp/test-list 
+    echo "### Running tests"
```
Comment (andfasano) [inline ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh:43]: Probably the above two ssh commands are not required

File: ci-operator/step-registry/baremetalds/e2e/baremetalds-e2e-workflow.yaml (hunk 1 above)
Comment (akiselev1) [reply ci-operator/step-registry/baremetalds/e2e/baremetalds-e2e-workflow.yaml:8]: Done

File: ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh (hunk 2 above)
Comment (akiselev1) [reply ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh:43]: Removed leftover debugs.

**Comment by stbenjam** [comment]
Ok just taking a look at this now - thanks for expanding the test coverage. Looks like e2e-metal-ipi is now about 2 hours. How do other platforms do it? I know there's different test suites, do they run all of them?

I'm concerned now that we're running all the time on the installer, mao and a few other places, having the increased test run time might make people unhappy.

**Comment by akiselev1** [comment]
Looking at baremetalds-e2e-test container logs: 
`56 pass, 0 skip (3m35s)`
Additional tests add about 3 min to run time. This is not the reason why it takes 2 hours now. Looks like dev-scripts setup takes longer in 4.6:
```
2020/07/15 22:00:31 Executing "e2e-metal-ipi-baremetalds-devscripts-setup"
2020/07/15 22:00:43 Container cp-secret-wrapper in pod e2e-metal-ipi-baremetalds-devscripts-setup completed successfully
2020/07/15 23:44:34 Copied 1.29MB of artifacts from e2e-metal-ipi-baremetalds-devscripts-setup to /logs/artifacts/e2e-metal-ipi/baremetalds-devscripts-setup
```
Other platforms are trying to execute the complete [Suite:openshift/conformance/parallel] with various degree of success. It may take 2:30-2:45 in case of AWS.

**Pull request merged**